includes/SQL.php

```diff
@@ -109,103 +109,123 @@ public function getPdoObj()
+                fputs ($fh, "\n".'=> Last erreur log');
+                fputs ($fh, "\n".var_export(error_get_last(), true));
+                fputs ($fh, "\n".'**************************************************');
+                fputs ($fh, "\n".'--------------------------------------------------');
+                fputs ($fh, "\n".'=> Debug Backtrace');
+                fputs ($fh, "\n".var_export($backtraces, true));
+                fputs ($fh, "\n".'**************************************************');
+                fputs ($fh, "\n".'--------------------------------------------------');
+                fputs ($fh, "\n".'=> Var dump $_REQUEST');
+                fputs ($fh, "\n".var_export($_REQUEST, true));
+                fputs ($fh, "\n".'--------------------------------------------------');
+                fputs ($fh, "\n".'=> Var dump $_SESSION');
+                fputs ($fh, "\n".var_export((isset($_SESSION) ? $_SESSION : array() ), true));
+                fputs ($fh, "\n".'--------------------------------------------------');
+                fputs ($fh, "\n".'=> Var dump $_SERVER');
+                fputs ($fh, "\n".var_export($_SERVER, true));
+                fputs ($fh, "\n".'**************************************************');
+                fclose ($fh);
+            }
+
+            @ob_clean();
+            // DONT USE GETTEXT ... KEEP THIS CODE WITHOUT REF ... THIS NEED TO BE A SAFE CODE !!!!
+            echo '<div style="margin: auto; width: 80%;">
+                    <h1>Une erreur est survenue ...</h1>
+                    <p>Pour aider la résolution de ce problème, veuillez fournir les informations suivantes :</p>
+                    <div>
+                        <div style="float:left;width: 230px;"><img src="'. IMG_PATH .'oops.png" style="width: 98%" /></div>
+                        <textarea style="width: 70%" rows="14">'.
+                        'login : ' .@$_SESSION['userlogin']."\n".
+                        'uri   : ' .preg_replace('/session=phpconges[a-z0-9]{32}&?/','',@$_SERVER['REQUEST_URI'])."\n".
+                        'dump  : ' .$dump_name. "\n"."\n".
+                        'file  : ' .$f['file']. "\n".
+                        'line  : ' .$f['line']. "\n".
+                        'fx    : $SQL->query'. "\n".
+                        'error : ' .$this->error. "\n".
+                        'sql   : ' .$query. "\n".
+                        '</textarea>
+                    </div>
+                </div>';
+            exit();
+        }
+        return $result;
+    }
+
+    public function quote( $escapestr )
+    {
+        return $this->escape_string( $escapestr );
+    }
+}
+
+
+class Database_MySQLi_Result extends \MySQLi_Result
+{
+    public function fetch()
+    {
+        return $this->fetch_assoc();
+    }
+
+    public function fetch_all()
+    {
+        $rows = array();
+        while($row = $this->fetch())
```

Comment:
Si tu mets directement `$this->fetch_assoc()`, tu peux virer la méthode du dessus